azurelinuxagent/common/utils/networkutil.py

```diff
@@ -156,18 +156,20 @@ def __get_firewalld_base_command(command):
         return ["firewall-cmd", "--permanent", "--direct", command, "ipv4"]
 
     @staticmethod
-    def __get_common_command_params(command, destination):
+    def __get_common_command_params(command, destination, pos):
+        if command == AddFirewallRules.INSERT_COMMAND:
+            return ["-t", "security", command, "OUTPUT", pos, "-d", destination, "-p", "tcp", "-m"]
         return ["-t", "security", command, "OUTPUT", "-d", destination, "-p", "tcp", "-m"]
 
     @staticmethod
-    def __get_common_accept_command_params(wait, command, destination, uid):
-        cmd = AddFirewallRules.__get_common_command_params(command, destination)
+    def __get_common_accept_command_params(wait, command, destination, uid, pos=""):
+        cmd = AddFirewallRules.__get_common_command_params(command, destination, pos)
```

Comment:
There's just way too many redirects for the `pos` column here which makes it really hard to follow (and think about cases if it can cause issues somewhere). Is there any way we can refactor this code to make it a bit more easier to understand? Maybe consider refactoring the code the same way `get_accept_tcp_rule` function is written? 

Additionally, since the `pos` is only needed for the `INSERT` commands, maybe consider creating a new function for it rather than hooking it up in the existing commands.

(I'm asking this just for clarity's sake because the current code is really hard to follow, not saying anything is wrong with the logic you applied here)